src/appdistribution/client.ts

```diff
@@ -69,32 +83,36 @@ export class AppDistributionClient {
   }
 
   async pollReleaseIdByHash(hash: string, retryCount = 0): Promise<string> {
-    try {
-      return await this.getReleaseIdByHash(hash);
-    } catch (err) {
+    const uploadStatus = await this.getUploadStatus(hash);
+    if (uploadStatus.status === UploadStatus.IN_PROGRESS) {
       if (retryCount >= AppDistributionClient.MAX_POLLING_RETRIES) {
-        throw new FirebaseError(`failed to find the uploaded release: ${err.message}`, { exit: 1 });
+        throw new FirebaseError(
+          "failed to fetch release information: polling timeout exceeded, please try again",
+          { exit: 1 }
+        );
       }
-
       await new Promise((resolve) =>
         setTimeout(resolve, AppDistributionClient.POLLING_INTERVAL_MS)
       );
-
       return this.pollReleaseIdByHash(hash, retryCount + 1);
+    } else if (uploadStatus.status === UploadStatus.SUCCESS) {
+      return uploadStatus.release.id;
+    } else {
+      throw new FirebaseError(`error processing the binary: ${uploadStatus.message}`);
```

Comment:
nit: `your binary`

Reply:
Can we append `(code: ${uploadStatus.errorCode})` on the end like we’re doing in the gradle plugin?